File: dm-test/src/test/java/ro/johann/dm/test/api/service/decision/transfer/AddAlternativeInput.java
package ro.johann.dm.test.api.service.decision.transfer;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Value;
import lombok.experimental.FieldDefaults;

import static lombok.AccessLevel.PRIVATE;

@Value
@Builder(builderClassName = "Builder")
@AllArgsConstructor
@FieldDefaults(level = PRIVATE)
public class AddAlternativeInput {
  String name;
}

File: dm-decision-service/src/main/kotlin/ro/johann/dm/decision/api/transfer/AddAlternativeInput.kt
package ro.johann.dm.decision.api.transfer

import ro.johann.dm.decision.domain.Alternative
import ro.johann.dm.decision.domain.Decision
import javax.validation.constraints.NotBlank

data class AddAlternativeInput(
  @field:NotBlank
  val name: String
) {
  fun toModel(decision: Decision): Alternative = Alternative(
    name = name,
    decision = decision
  )
}